app/tests/payment/test_paid_event_integration.py

```diff
@@ -0,0 +1,164 @@
+    Other subgroup members can create events where event.organizer is their group or None.
+    Leaders of committees and interest groups should be able to
+    update events where event.organizer is their group or None.
+    """
+
+    organizer = Group.objects.get_or_create(name="HS", type=GroupType.BOARD)[0]
+    client = get_api_client(user=admin_user)
+    data = get_paid_event_data(organizer=organizer.slug)
+
+    response = client.post(API_EVENTS_BASE_URL, data)
+    created_event = Event.objects.get(title=data["title"])
+    paid_event_information = PaidEvent.objects.get(event=created_event)
+
+    assert response.status_code == 201
+    assert created_event.is_paid_event
+    assert paid_event_information.price == data["paid_information"]["price"]
+    assert (
+        float(response.data["paid_information"]["price"])
+        == data["paid_information"]["price"]
+    )
+
+
+@pytest.mark.django_db
+def test_create_paid_event_without_price_as_admin(admin_user):
+    """
+    HS and Index members should be able to create paid events no matter which organizer is selected.
+    Other subgroup members can create events where event.organizer is their group or None.
+    Leaders of committees and interest groups should be able to
+    update events where event.organizer is their group or None.
+    """
+
+    organizer = Group.objects.get_or_create(name="HS", type=GroupType.BOARD)[0]
+    client = get_api_client(user=admin_user)
+    data = get_paid_event_without_price_data(organizer=organizer.slug)
+
+    response = client.post(API_EVENTS_BASE_URL, data)
+    created_event = Event.objects.get(title=data["title"])
+    paid_event_information = PaidEvent.objects.get(event=created_event)
+
+    assert response.status_code == 201
+    assert created_event.is_paid_event
+    assert paid_event_information.price == 0.00
+    assert float(response.data["paid_information"]["price"]) == 0.00
+
+
+@pytest.mark.django_db
+def test_update_paid_event_as_admin(admin_user):
+    """
+    HS and Index members should be able to update all paid events.
+    Other subgroup members can update paid events where event.organizer is their group or None.
+    Leaders of committees and interest groups should be able to
+    update events where event.organizer is their group or None.
+    """
+
+    new_event_price = 100.00
+    paid_event = PaidEventFactory(price=0.00)
+    event = paid_event.event
+    organizer = Group.objects.get_or_create(name="HS", type=GroupType.BOARD)[0]
+    client = get_api_client(user=admin_user)
+    url = get_events_url_detail(event)
+    data = get_paid_event_data(organizer=organizer.slug)
+
+    response = client.put(url, data)
+    event.refresh_from_db()
+
+    assert response.status_code == 200
+    assert event.paid_information.price == new_event_price
+    assert float(response.data["paid_information"]["price"]) == new_event_price
+
+
+# This test does not work while using a celery task
```

Comment:
Finne en løsning for dette

Reply:
Jeg har fått testet dette via frontend. Men skal få sett på det. Det kan hende at nå som jeg fikk fikset det med celery at testene også vil funke, siden  tasken ikke kjører med en gang.